src/alt/store/index.js

```diff
@@ -53,7 +53,13 @@ export function createStoreConfig(globalConfig, StoreModel) {
         return state
       }
     },
-    setState: fn.assign
+    setState(currentState, nextState) {
+      if (Object.prototype.toString.call(nextState) === '[object Object]') {
```

Comment:
Maybe pull this out into a `fn.isPlainObject`? You use it e.g. at https://github.com/goatslacker/alt/blob/44d581401bcf03d531537a70285edf28fcebfc4b/src/alt/store/index.js#L50 as well.